solr/modules/llm/src/test/org/apache/solr/llm/TestLlmBase.java

```diff
@@ -0,0 +1,297 @@
+  protected static Path tmpConfDir;
+
+  public static final String MODEL_FILE_NAME = "_schema_embedding-model-store.json";
+  protected static final String COLLECTION = "collection1";
+  protected static final String CONF_DIR = COLLECTION + "/conf";
+
+  protected static Path embeddingModelStoreFile = null;
+
+  protected static String IDField = "id";
+  protected static String stringField = "string_field";
+  protected static String vectorField = "vector";
+  protected static String vectorField2 = "vector2";
+  protected static String vectorFieldByteEncoding = "vector_byte_encoding";
+
+  protected static void setuptest(boolean bulkIndex) throws Exception {
+    setuptest("solrconfig-llm.xml", "schema.xml");
+    if (bulkIndex) prepareIndex();
+  }
+
+  protected static void setupPersistenttest(boolean bulkIndex) throws Exception {
+    setupPersistentTest("solrconfig-llm.xml", "schema.xml");
+    if (bulkIndex) prepareIndex();
+  }
+
+  public static ManagedEmbeddingModelStore getManagedModelStore() {
+    try (SolrCore core = solrClientTestRule.getCoreContainer().getCore(DEFAULT_TEST_CORENAME)) {
+      return ManagedEmbeddingModelStore.getManagedModelStore(core);
+    }
+  }
+
+  protected static void setupTestInit(String solrconfig, String schema, boolean isPersistent)
+      throws Exception {
+    tmpSolrHome = createTempDir();
+    tmpConfDir = tmpSolrHome.resolve(CONF_DIR);
+    tmpConfDir.toFile().deleteOnExit();
+    PathUtils.copyDirectory(TEST_PATH(), tmpSolrHome.toAbsolutePath());
+
+    final Path mstore = tmpConfDir.resolve(MODEL_FILE_NAME);
+
+    if (isPersistent) {
+      embeddingModelStoreFile = mstore;
+    }
+
+    if (Files.exists(mstore)) {
+      if (log.isInfoEnabled()) {
+        log.info("remove model store config file in {}", mstore.toAbsolutePath());
+      }
+      Files.delete(mstore);
+    }
+    if (!solrconfig.equals("solrconfig-llm.xml")) {
+      Files.copy(
+          tmpSolrHome.resolve(CONF_DIR).resolve(solrconfig),
+          tmpSolrHome.resolve(CONF_DIR).resolve("solrconfig-llm.xml"));
+    }
+    if (!schema.equals("schema.xml")) {
+      Files.copy(
+          tmpSolrHome.resolve(CONF_DIR).resolve(schema),
+          tmpSolrHome.resolve(CONF_DIR).resolve("schema.xml"));
+    }
+
+    System.setProperty("managed.schema.mutable", "true");
+  }
+
+  public static void setuptest(String solrconfig, String schema) throws Exception {
+
+    setupTestInit(solrconfig, schema, false);
+    System.setProperty("enable.update.log", "false");
+
+    createJettyAndHarness(
+        tmpSolrHome.toAbsolutePath().toString(), solrconfig, schema, "/solr", true, null);
+  }
+
+  public static void setupPersistentTest(String solrconfig, String schema) throws Exception {
+
+    setupTestInit(solrconfig, schema, true);
+
+    createJettyAndHarness(
+        tmpSolrHome.toAbsolutePath().toString(), solrconfig, schema, "/solr", true, null);
+  }
+
+  protected static void aftertest() throws Exception {
```

Comment:
this part is coming from the Learning To Rank module at least, it cleans some directories created during the test, I admit I didn't invest much time here to see if that's the best way of testing, but given the similar scenario in Learning To Rank, I thought it would be ok to start from there